src/KeyValueStore/keyvaluestore_server.cc

```diff
@@ -55,14 +84,22 @@ void RunServer() {
 
   LOG(INFO) << "Server listening on " << server_address;
 
+  // register signal SIGINT and signal handler
+  signal(SIGINT, signalHandler);
+  shutdownHandler = [&](int signal) {
+    std::cout << "Server shutdown... " <<std::endl;
+    service.store();
```

Comment:
What if multiple clients store data to the same file at the same time?  This might be a rare case.

Reply:
You are right, this is a case. However, this signalHandler called in the kvstore_server, which is just run once and has nothing to do with clients. When kvstore_server is running, all the data is running in memory, and all clients interacted with kvstore_server and in-memory data. When kvstore_server is terminated, signalHanlder will store the data into the local file, and no clients could connect with the server and no clients could access the data. Thus, in another word, multiple clients just access the in-memory kv_store map. When kvstore_store is terminated, all data will be written to the file.